src/entrypoint.sh

```diff
@@ -6,10 +6,16 @@ set -x
 # shellcheck source=src/sync_common.sh
 source sync_common.sh
 
-[ -z "${GITHUB_TOKEN}" ] && {
-  err "Missing input 'github_token: \${{ secrets.GITHUB_TOKEN }}'.";
-  exit 1;
-};
+if [[ -z "${GITHUB_TOKEN}" && -z "${SOURCE_REPO_GITHUB_TOKEN}" ]]; then
+# TODO
+    err "Missing input 'github_token: \${{ secrets.GITHUB_TOKEN }}'.";
+    exit 1;
+fi
+
+# if [[ -z "${SOURCE_REPO_GITHUB_TOKEN}" ]]; then
+#   debug "Missing input 'source_repo_github_token: \${{ input.source_repo_github_token }}'. Using github_token as default."
+#   SOURCE_REPO_GITHUB_TOKEN="${GITHUB_TOKEN}"
```

Comment:
This will not work on my side and make `source_repo_github_token` mandatory. Is run a setup for example where I have one app to interact with both repositories.

Reply:
this is commented. Will be removed afterwards, sorry for not cleaning up.
There are some further parts not cleaned up (yet) with TODO comments.
Also the documentation must be updated. Just wanted to make sure that it is working.

So please ignore the commented parts (for now). (sorry for that)

Reply:
no worries - but you miss understand my comment: I need `SOURCE_REPO_GITHUB_TOKEN="${GITHUB_TOKEN}"` in my setup, otherwise I need to reconfigure my actions. 

But if I re-add `SOURCE_REPO_GITHUB_TOKEN="${GITHUB_TOKEN}"` to the code the rest looks working for now.

Reply:
ah ok, but in your case the `GITHUB_TOKEN` is the token created for the Github app, right?
Then maybe it is a bit more generic to have in if statement line 69-79 an else path like

```bash
else
   gh auth login --git-protocol "https" --hostname "${SOURCE_REPO_HOSTNAME}" --with-token <<< "${GITHUB_TOKEN}"
    gh auth setup-git --hostname "${SOURCE_REPO_HOSTNAME}"
    # git config --global "credential.https://${SOURCE_REPO_HOSTNAME}.helper" "!gh auth git-credential"
    gh auth status
```

Reply:
> ah ok, but in your case the GITHUB_TOKEN is the token created for the Github app, right?
yes - the GITHUB_TOKEN is generated by an app to handle the repo access and was also used for the checkout. But it would be also the same if you would re-use the gh actions app with the right permissions.